dali/operators/reader/loader/filesystem.cc

```diff
@@ -127,54 +132,105 @@ std::vector<std::string> list_files(const std::string &parent_dir,
   return files;
 }
 
-vector<std::pair<string, int>> traverse_directories(const std::string &file_root,
-                                                    const std::vector<std::string> &filters,
-                                                    bool case_sensitive_filter,
-                                                    const std::vector<std::string> &dir_filters) {
-  std::vector<std::string> subdirs;
+
+#if AWSSDK_ENABLED
+
+vector<FileLabelEntry> s3_traverse_directories(const std::string &file_root,
+                                               const TraverseDirectoriesOptions &opts) {
   bool is_s3 = starts_with(file_root, "s3://");
-  if (is_s3)
-    DALI_FAIL("This version of DALI was not built with AWS S3 storage support.")
-  subdirs = list_subdirectories(file_root, dir_filters, case_sensitive_filter);
+  assert(is_s3);
```

Comment:
```suggestion
  assert(starts_with(file_root, "s3://"));
```
otherwise a build without asserts may complain about a variable initialized but never used.